### ISSUE TITLE
Char data type

### ISSUE THREAD

**Issue by harshnambiar** [noir-lang/noir#1636]
### Problem

As we all know, Char is a fundamental type for any language and without it, a lot of things that can be done with a few lines of code, cannot be done with a few lines of code. For example, it will be quite useful in all numerical to textual conversion, random number generation, alternative hashing functions and so on. Hence it would be great if this can be delivered as a priority.

### Happy Case

The implementation should mirror those in other languages .

_**let mut c: Char = 's';
let mut x = "ftring";

x = x.replace('f', c);**_

Something like that at least 

### Alternatives Considered

Unfortunately, this is non-negotiable to ensure the survival of the language in the longer term.

### Would you like to submit a PR for this Issue?

No

**Comment by harshnambiar** [comment]
> Hi there, @harshnambiar. I would consider `Unfortunately, this is non-negotiable to ensure the survival of the language in the longer term.` to be the wrong attitude. However a char type seems simple enough, but your string mutation example is more involved.

Sorry if saying a language without a fundamental data type might not survive felt like a wrong attitude to have.

**Comment by ludamad** [comment]
> > Hi there, @harshnambiar. I would consider `Unfortunately, this is non-negotiable to ensure the survival of the language in the longer term.` to be the wrong attitude. However a char type seems simple enough, but your string mutation example is more involved.
> 
> Sorry if saying a language without a fundamental data type might not survive felt like a wrong attitude to have.

It's no worries, but I do want to point out how I'm sure others in ZK DSL space read this issue. We consider that it's a nice-to-have data type, anything more needs convincing. This is enough to implement it! So no worries. But if you want to further convince concepts such as 'fundamental', 'ASAP', 'non-negotiable' you do need to understand *our* thinking, not just your own

**Comment by ludamad** [comment]
As for my thinking: Yes string primitives are eventually something we need but we are still iterating on fundamental concepts such as how loops compile, the ability to run unconstrained code, etc. For now it's enough for us to know what we do allows computation to be done efficiently, these features we eventually will fill out, but just want to level set what 'ASAP' might mean

**Comment by harshnambiar** [comment]
> > > Hi there, @harshnambiar. I would consider `Unfortunately, this is non-negotiable to ensure the survival of the language in the longer term.` to be the wrong attitude. However a char type seems simple enough, but your string mutation example is more involved.
> > 
> > 
> > Sorry if saying a language without a fundamental data type might not survive felt like a wrong attitude to have.
> 
> It's no worries, but I do want to point out how I'm sure others in ZK DSL space read this issue. We consider that it's a nice-to-have data type, anything more needs convincing. This is enough to implement it! So no worries. But if you want to further convince concepts such as 'fundamental', 'ASAP', 'non-negotiable' you do need to understand _our_ thinking, not just your own

I think at the end of the day, both of us want what's best for Noir. Even if what we think is "fundamental" differs, I totally wish you the best, and if Noir does end up without things like string handling, that's totally ok! I was actually told to file this issue only after discussing with the Noir team members itself, who agreed the need for a Char type. But if Noir does intend to be some particular niche and not implement some common constructs seen in other languages, that's all great but hopefully they could mention it so somewhere on the official blog, like to avoid any misunderstandings in the future. Cheers :)

**Comment by ludamad** [comment]
https://book.starknet.io/chapter_2/strings.html I believe the other ZK DSL with normal programming syntax is in a similar place with their strings
